### ISSUE TITLE
Drastically simplify permissions

### ISSUE THREAD

**Issue by joshsmith** [code-corps/deprecated-code-corps-rails-api#175]
After a lot of thought, I think we need to drastically simply permissions to accelerate a V1 of the product.

Essentially we are going to get rid of teams and project-level permissioning in favor of a radically reduced permissioning model that favors users only having membership in an organization, and having access rights to all projects inside of that organization.

This means:
- [x] Remove `Team` and `TeamProject` models
- [x] Remove `Contributor` model
- [x] Remove `TeamMembership` model
- [x] Remove contributors count on projects
- [x] Change `OrganizationMembership` `role` from `regular` to `contributor`
- [x] Add `pending` to `OrganizationMembership` `role`
- [x] Rewrite `PostPolicy` to use `OrganizationMembership` instead of `Contributor`
- [x] Rewrite `ProjectPolicy` to use `OrganizationMembership` instead of `Contributor`
- [x] Write `GithubRepositoryPolicy` spec
- [x] Rewrite `GithubRepositoryPolicy` to use `OrganizationMembership` instead of `Contributor`
- [x] Rewrite Posts API spec to use `OrganizationMembership` instead of `Contributor`
- [x] Replace repositories API spec `contributor` with `organization_membership`
- [x] Rewrite Users API spec to not use `contributor`
- [x] Open `idea` post types to everyone (not just contributors)
- [x] Remove polymorphism from projects and make `belongs_to :organization`
- [x] Organization admins can create projects
- [x] Organization contributors can create tasks, close tasks, etc.

We've spent a lot of time dealing with permissions that I don't know are going to be needed for another a year in the app. We can get by with this right up until we can't. And then we can build it. For now, we should focus on shipping what matters.

**Comment by begedin** [comment]
I'm all for simplifying this.

Just to make sure I understand.

User's will have to belong to an organization to create and have access to projects in that organization. I'm not completely clear on whether there are any roles within an OrganizationMembership, but my guess is, contributors can access projects, while some higher role can also create them?

Just to make sure I understand this

A user on their own does nothing. They just have a profile with skills, etc. 

An organization is the entity that owns projects.

To take part in a project, a user requests access to an organization

That would mean there would be the following levels of organization access
- pending (user requested access to be member of organization)
- contributor (user got access)
- some higher level(s) - user can also create projects, approve pending OrganizationMembership requests, kick people out, etc., handle posts, etc.

Basically **what the user can do with projects and posts for that project depends on organization membership level for that user and the organization the project belongs to**

I like that.

**Comment by scarfdestroyer** [comment]
We are all driving somewhere right now but, what you said was pretty much all correct and Josh says he only wants to add more granular permissioning if users demand it.

**Comment by begedin** [comment]
All of the subtasks here seem to be done, but regarding opening idea posts to everyone, I did so for all registered users, regardless of level or membership, but not unregistered users, since we do not allow that anywhere. I'm guessing that's ok?

**Comment by joshsmith** [comment]
That's perfect.
